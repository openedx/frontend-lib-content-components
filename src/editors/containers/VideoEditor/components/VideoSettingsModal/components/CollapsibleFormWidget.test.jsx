import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { formatMessage } from '../../../../../../testUtils';
import { CollapsibleFormWidget } from './CollapsibleFormWidget';

describe('CollapsibleFormWidget', () => {
  const props = {
    isError: false,
    subtitle: 'SuBTItle',
    title: 'tiTLE',
    // inject
    intl: { formatMessage },
  };
  describe('render', () => {
    const testContent = (<p>Some test string</p>);
    test('snapshots: renders as expected with default props', () => {
      expect(
        shallow(<CollapsibleFormWidget {...props}>{testContent}</CollapsibleFormWidget>).snapshot,
      ).toMatchSnapshot();
    });
    test('snapshots: renders with open={true} when there is error', () => {
      expect(
        shallow(<CollapsibleFormWidget {...props} isError>{testContent}</CollapsibleFormWidget>).snapshot,
      ).toMatchSnapshot();
    });
  });
});
