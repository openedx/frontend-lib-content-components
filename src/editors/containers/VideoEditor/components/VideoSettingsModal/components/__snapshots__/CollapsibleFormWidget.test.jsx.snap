// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleFormWidget render snapshots: renders as expected with default props 1`] = `
<Advanced
  className="collapsible-card rounded mx-4 my-3 px-4 text-primary-500"
  defaultOpen={true}
>
  <Trigger
    className="collapsible-trigger d-flex border-0 align-items-center pt-4 p-0"
    style={
      {
        "justifyContent": "unset",
      }
    }
  >
    <Visible
      className="p-0 pb-3"
      whenClosed={true}
    >
      <div
        className="d-flex flex-column flex-grow-1"
      >
        <div
          className="d-flex flex-grow-1 w-75 x-small"
        >
          tiTLE
        </div>
        <div
          className=" mb-4 mt-3"
        >
          SuBTItle
        </div>
      </div>
      <div
        className="d-flex flex-row align-self-start"
      >
        <IconButton
          alt="Expand"
          iconAs="Icon"
          variant="dark"
        />
      </div>
    </Visible>
    <Visible
      whenOpen={true}
    >
      <div
        className="d-flex flex-grow-1 w-75 x-small"
      >
        tiTLE
      </div>
      <div
        className="align-self-start"
      >
        <IconButton
          alt="Collapse"
          iconAs="Icon"
          variant="dark"
        />
      </div>
    </Visible>
  </Trigger>
  <Body
    className="collapsible-body rounded px-0  pb-4"
  >
    <p>
      Some test string
    </p>
  </Body>
</Advanced>
`;

exports[`CollapsibleFormWidget render snapshots: renders with open={true} when there is error 1`] = `
<Advanced
  className="collapsible-card rounded mx-4 my-3 px-4 text-primary-500"
  defaultOpen={true}
  open={true}
>
  <Trigger
    className="collapsible-trigger d-flex border-0 align-items-center pt-4 p-0"
    style={
      {
        "justifyContent": "unset",
      }
    }
  >
    <Visible
      className="p-0 pb-3"
      whenClosed={true}
    >
      <div
        className="d-flex flex-column flex-grow-1"
      >
        <div
          className="d-flex flex-grow-1 w-75 x-small"
        >
          tiTLE
        </div>
        <div
          className=" mb-4 mt-3"
        >
          SuBTItle
        </div>
      </div>
      <div
        className="d-flex flex-row align-self-start"
      >
        <Icon
          className="alert-icon"
        />
        <IconButton
          alt="Expand"
          iconAs="Icon"
          variant="dark"
        />
      </div>
    </Visible>
    <Visible
      whenOpen={true}
    >
      <div
        className="d-flex flex-grow-1 w-75 x-small"
      >
        tiTLE
      </div>
      <div
        className="align-self-start"
      >
        <IconButton
          alt="Collapse"
          iconAs="Icon"
          variant="dark"
        />
      </div>
    </Visible>
  </Trigger>
  <Body
    className="collapsible-body rounded px-0  pb-4"
  >
    <p>
      Some test string
    </p>
  </Body>
</Advanced>
`;
